### PR TITLE
Move rb_id2str into new LLDB format

### PR DESCRIPTION
**Pull request by eightbitraptor**
Old command has been left in place and aliased as `old_rbid2str` to aid with debugging.

### CODE REVIEW

File: misc/lldb_rb/commands/rb_id2str_command.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import lldb
+
+from lldb_rb.constants import *
+from lldb_rb.rb_base_command import RbBaseCommand
+
+class RbID2StrCommand(RbBaseCommand):
+    program = "rb_id2str"
+
+    help_string = "convert and print a Ruby ID to s char * and print it to the lldb console"
```
Comment (peterzhu2118) [inline misc/lldb_rb/commands/rb_id2str_command.py:9]: This string is confusing.

```suggestion
    help_string = "convert a Ruby ID to a C string and print it to the LLDB console"
```

File: misc/lldb_rb/commands/rb_id2str_command.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        id_val = self.frame.EvaluateExpression(command).GetValueAsUnsigned()
+        num = self.rb_id_to_serial(id_val)
+
+        last_id = global_symbols.GetChildMemberWithName("last_id").GetValueAsUnsigned()
+        ID_ENTRY_SIZE = 2
+        ID_ENTRY_UNIT = int(self.target.FindFirstGlobalVariable("ID_ENTRY_UNIT").GetValue())
+
+        ids = global_symbols.GetChildMemberWithName("ids")
+
+        if (num <= last_id):
```
Comment (peterzhu2118) [inline misc/lldb_rb/commands/rb_id2str_command.py:23]: Nit: don't need the parenthesis in Python.

File: misc/lldb_rb/commands/rb_id2str_command.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            idx = num // ID_ENTRY_UNIT
+            ary = self.rb_ary_entry(target, ids, idx, result)
+            pos = (num % ID_ENTRY_UNIT) * ID_ENTRY_SIZE
+            id_str = self.rb_ary_entry(target, ary, pos, result)
+
+            self.lldb_inspect(debugger, target, result, id_str)
+
+    def rb_id_to_serial(self, id_val):
+        print(globals())
```
Comment (peterzhu2118) [inline misc/lldb_rb/commands/rb_id2str_command.py:32]: Is this `print` leftover from debugging?

File: misc/lldb_rb/rb_base_command.py [hunk 4]
```diff
@@ -67,3 +67,191 @@ def _append_command_output(self, debugger, command, result):
         result.Clear()
         result.write(output1)
         result.write(output2)
+
+    def lldb_inspect(debugger, target, result, val):
```
Comment (peterzhu2118) [inline misc/lldb_rb/rb_base_command.py:71]: Is there some way to get rid of the duplicated `lldb_inspect` in `lldb_cruby.py` and use this instead?

Comment (eightbitraptor) [reply misc/lldb_rb/rb_base_command.py:70]: I'm not super concerned about this at the moment. Eventually there shouldn't be anything left in this file besides the object hierarchy initialisation when everything gets ported across anyway.

**Pull request merged**